Makefile

```diff
@@ -74,12 +74,19 @@ check-no-changes:
 
 .PHONY: test
 test:  ## - Run all tests
+	@mkdir -p build
 	@$(MAKE) test-unit 
 	# @$(MAKE) test-int
+	@$(MAKE) junit-report
 
 .PHONY: test-unit 
 test-unit: ## - Run unit tests only
-	@go test -v -race ./...
+	@go test -v -race ./... | tee build/test-unit.out
```

Comment:
you'd need to set pipefail for bash

```
.SHELLFLAGS = -eo pipefail
```

Reply:
I was somehow skeptical when I wrote it since I wanted to delegate the junit reporting to be the one driving the build status. But there might be some corner cases that it's worth to use the shellflags

Reply:
the thing is junit will mark the build as unstable, not as failed